extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/cache/InfinispanGetWrapper.java

```diff
@@ -0,0 +1,76 @@
+package io.quarkus.infinispan.client.runtime.cache;
+
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.commons.CacheException;
+
+import io.smallrye.mutiny.Uni;
+
+public class InfinispanGetWrapper {
+    final RemoteCache cache;
+    final Map<Object, ReentrantLock> synchronousGetLocks;
```

Comment:
This whole class needs to be rewritten to be non blocking. All of the Cache operations in `getFromInfinispan` are blocking.
I would suggest changing this to be `Map<Object, CompletionStage<Object>>` to allow for that.